File: extensions/text.py
from pathlib import Path

from wutilerror import check_types
import debugutils

log = debugutils.log
mark = debugutils.mark_time


class Extension:
    def __init__(self):
        self.name = "text"
        self.desc = "An extension to output little tidbits of text that you use regularly."
        self.args = ["id"]
        self.short = "txt"
        self.requires_window = False

    def main(self, id="empty"):
        check_types(
            id=(id, str, True)
        )

        mark("text start", source="text")

        text_path = Path.home() / ".wutil" / "text" / f"{id}.txt"

        if not text_path.exists():
            print(f"Text '{id}' not found.")
            log(f"text not found. id={id}", source="text")
            return
        text = text_path.read_text(encoding="utf-8")

        print(text)

        log(f"text outputted. id={id}", source="text")

File: extensions/text.go
package main

//wutil:name text
//wutil:short txt
//wutil:desc An extension to output little tidbits of text that you use regularly.
//wutil:args id
//wutil:requires_window false

import (
	"fmt"
	"os"
	"path/filepath"

	"windowutil/pkg/wutilext"
)

func main() {
	id := "empty"
	if len(os.Args) > 1 {
		id = os.Args[1]
	}

	wutilext.Mark("text start", "text")
	home, err := os.UserHomeDir()
	if err != nil {
		fmt.Println("Text home directory not found.")
		wutilext.Log(fmt.Sprintf("text failed to resolve home directory: %s", err), "text")
		return
	}
	textPath := filepath.Join(home, ".wutil", "text", id+".txt")
	data, err := os.ReadFile(textPath)
	if err != nil {
		fmt.Printf("Text '%s' not found.\n", id)
		wutilext.Log(fmt.Sprintf("text not found. id=%s", id), "text")
		return
	}
	fmt.Print(string(data))
	if len(data) > 0 && data[len(data)-1] != '\n' {
		fmt.Println()
	}
	wutilext.Log(fmt.Sprintf("text outputted. id=%s", id), "text")
}
